Packages/io.chainsafe.web3-unity/Runtime/Resources/ABIContractClassTemplate.txt

```diff
@@ -49,27 +50,63 @@ namespace ChainSafe.Gaming.Evm.Contracts.Custom
                 return;
             if(!Subscribed)
                 return;
+                
+            if(Application.platform == RuntimePlatform.WebGLPlayer)
+               return;
             Subscribed = false;
+            try
+            {
             {EVENT_UNSUBSCRIPTION}
-            _webSocketClient?.Dispose();
+            
+             if (_webSocketClient != null)
+                await _webSocketClient.StopAsync();
+            }catch(Exception e)
+            {
+                Debug.LogError("Caught an exception whilst unsubscribing from events\n" + e.Message);
+            }
         }
         
-        public async ValueTask Init()
+        public async ValueTask InitAsync()
         {
             if(Subscribed)
                 return;
-                
+            Subscribed = true;
+
             if(string.IsNullOrEmpty(WebSocketUrl))
             {
                 Debug.LogWarning($"WebSocketUrl is not set for this class. Event Subscriptions will not work.");
                 return;
             }
+
+            
            
-            _webSocketClient ??= new StreamingWebSocketClient(WebSocketUrl);
-            await _webSocketClient.StartAsync();
-            Subscribed = true;
+            try
+            {
+                if(Application.platform == RuntimePlatform.WebGLPlayer)
+                {
+                   Debug.LogWarning("WebGL Platform is currently not supporting event subscription");
```

Comment:
even with async utils doesn't work?

Reply:
No because this uses native http websockets from .net 😢